src/test/java/jenkins/plugins/logstash/LogstashWriterTest.java

```diff
@@ -0,0 +1,299 @@
+package jenkins.plugins.logstash;
+
+import hudson.model.AbstractBuild;
+import hudson.model.Project;
+import hudson.model.Result;
+import hudson.tasks.test.AbstractTestResultAction;
+import jenkins.plugins.logstash.persistence.BuildData;
+import jenkins.plugins.logstash.persistence.LogstashIndexerDao;
+import jenkins.plugins.logstash.persistence.LogstashIndexerDao.IndexerType;
+import net.sf.json.JSONObject;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.GregorianCalendar;
+
+import static org.hamcrest.core.StringContains.containsString;
+import static org.hamcrest.collection.IsIterableContainingInOrder.contains;
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.*;
```

Comment:
You have some unused imports that you can clean up:

java.util.Collection
org.hamcrest.collection.IsIterableContainingInOrder.contains
org.junit.Assert.assertThat